src/e2ee/sharedKeyManagement.ts

```diff
@@ -17,47 +17,62 @@ limitations under the License.
 import { useEffect, useMemo } from "react";
 
 import { useEnableE2EE } from "../settings/useSetting";
-import { useLocalStorage } from "../useLocalStorage";
+import { setLocalStorageItem, useLocalStorage } from "../useLocalStorage";
 import { useClient } from "../ClientContext";
 import { useUrlParams } from "../UrlParams";
 import { widget } from "../widget";
 
 export const getRoomSharedKeyLocalStorageKey = (roomId: string): string =>
   `room-shared-key-${roomId}`;
 
-const useInternalRoomSharedKey = (
-  roomId: string
-): [string | null, (value: string) => void] => {
-  const key = useMemo(() => getRoomSharedKeyLocalStorageKey(roomId), [roomId]);
+const useInternalRoomSharedKey = (roomId: string): string | null => {
+  const key = getRoomSharedKeyLocalStorageKey(roomId);
   const [e2eeEnabled] = useEnableE2EE();
-  const [roomSharedKey, setRoomSharedKey] = useLocalStorage(key);
+  const roomSharedKey = useLocalStorage(key)[0];
 
-  return [e2eeEnabled ? roomSharedKey : null, setRoomSharedKey];
+  return e2eeEnabled ? roomSharedKey : null;
 };
 
-const useKeyFromUrl = (roomId: string): string | null => {
+/**
+ * Extracts the room password from the URL if one is present, saving it in localstorage
+ * and returning it in a tuple with the corresponding room ID from the URL.
+ * @returns A tuple of the roomId and password from the URL if the URL has both,
+ *          otherwise [undefined, undefined]
+ */
+const useKeyFromUrl = (): [string, string] | [undefined, undefined] => {
   const urlParams = useUrlParams();
-  const [e2eeSharedKey, setE2EESharedKey] = useInternalRoomSharedKey(roomId);
 
   useEffect(() => {
```

Comment:
why exactly do we still need the effect?
What would happen if we do:
```js
const useKeyFromUrl = (): [string, string] | [undefined, undefined] => {
  if (urlParams.roomId && urlParams.password) {
   setLocalStorageItem(
      getRoomSharedKeyLocalStorageKey(urlParams.roomId),
      urlParams.password
    );
    return [urlParams.roomId, urlParams.password];
  }
   return [undefined, undefined];
}

Reply:
It would grab the key out of the url on every re-render: writing to localsotrage is fairly fast, but probably not fast enough I'd want to block renders on it, and obviously we don't need to do it every time.